### ISSUE TITLE
enterprise cluster: batch operation with rest call throw an exception with message "No thread local value in scope for proxy..."

### ISSUE THREAD

**Issue by mmeloni** [neo4j/neo4j#7648]
- Neo4j version: 3.0.3
- Operating system: debian 8
- API/Driver: rest batch operation 
- To reproduce install a clean cluster ha 
- Actual behavior: throw an exception with message "No thread local value in scope for proxy of class"

We have installed a clean enterprise cluster and try a batch operation with rest interface.

Ex:

```
curl -H "Content-Type: application/json" -d '[ {"method" : "PUT","to" : "/node/496/properties","body" : {"age" : 1},"id" : 0}, {"method" : "GET","to" : "/node/496","id" : 1}, {"method" : "POST","to" : "/node","body" : {"age" : 1},"id" : 2}, {"method" : "POST","to" : "/node","body" : {"age" : 1},"id" : 3}]' http://127.0.0.1:7474/db/data/batch
```

Actually we have this exception:

```
{
  "message" : "No thread local value in scope for proxy of class com.sun.proxy.$Proxy72",
  "exception" : "IllegalStateException",
  "fullname" : "java.lang.IllegalStateException",
  "stackTrace" : [ "com.sun.proxy.$Proxy72.getRemoteHost(Unknown Source)", "org.neo4j.server.rest.web.InternalJettyServletRequest.getRemoteHost(InternalJettyServletRequest.java:222)", "org.neo4j.server.web.AsyncRequestLog.log(AsyncRequestLog.java:69)", "org.neo4j.server.web.Jetty9WebServer.invokeDirectly(Jetty9WebServer.java:323)", "org.neo4j.server.rest.batch.NonStreamingBatchOperations.invoke(NonStreamingBatchOperations.java:56)", "org.neo4j.server.rest.batch.BatchOperations.performRequest(BatchOperations.java:215)", "org.neo4j.server.rest.batch.BatchOperations.parseAndPerform(BatchOperations.java:183)", "org.neo4j.server.rest.batch.NonStreamingBatchOperations.performBatchJobs(NonStreamingBatchOperations.java:49)", "org.neo4j.server.rest.web.BatchOperationService.batchProcess(BatchOperationService.java:155)", "org.neo4j.server.rest.web.BatchOperationService.performBatchOperations(BatchOperationService.java:85)", "java.lang.reflect.Method.invoke(Method.java:498)", "org.neo4j.server.rest.transactional.TransactionalRequestDispatcher.dispatch(TransactionalRequestDispatcher.java:144)", "org.neo4j.server.rest.web.CollectUserAgentFilter.doFilter(CollectUserAgentFilter.java:69)", "java.lang.Thread.run(Thread.java:745)" ],
  "errors" : [ {
    "code" : "Neo.DatabaseError.General.UnknownError",
    "stackTrace" : "java.lang.IllegalStateException: No thread local value in scope for proxy of class com.sun.proxy.$Proxy72\n\tat com.sun.jersey.server.impl.ThreadLocalInvoker.invoke(ThreadLocalInvoker.java:93)\n\tat com.sun.proxy.$Proxy72.getRemoteHost(Unknown Source)\n\tat org.neo4j.server.rest.web.InternalJettyServletRequest.getRemoteHost(InternalJettyServletRequest.java:222)\n\tat org.neo4j.server.web.AsyncRequestLog.log(AsyncRequestLog.java:69)\n\tat org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:113)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)\n\tat org.neo4j.server.web.Jetty9WebServer.invokeDirectly(Jetty9WebServer.java:323)\n\tat org.neo4j.server.rest.batch.NonStreamingBatchOperations.invoke(NonStreamingBatchOperations.java:56)\n\tat org.neo4j.server.rest.batch.BatchOperations.performRequest(BatchOperations.java:215)\n\tat org.neo4j.server.rest.batch.BatchOperations.parseAndPerform(BatchOperations.java:183)\n\tat org.neo4j.server.rest.batch.NonStreamingBatchOperations.performBatchJobs(NonStreamingBatchOperations.java:49)\n\tat org.neo4j.server.rest.web.BatchOperationService.batchProcess(BatchOperationService.java:155)\n\tat org.neo4j.server.rest.web.BatchOperationService.performBatchOperations(BatchOperationService.java:85)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.lang.reflect.Method.invoke(Method.java:498)\n\tat com.sun.jersey.spi.container.JavaMethodInvokerFactory$1.invoke(JavaMethodInvokerFactory.java:60)\n\tat com.sun.jersey.server.impl.model.method.dispatch.AbstractResourceMethodDispatchProvider$ResponseOutInvoker._dispatch(AbstractResourceMethodDispatchProvider.java:205)\n\tat com.sun.jersey.server.impl.model.method.dispatch.ResourceJavaMethodDispatcher.dispatch(ResourceJavaMethodDispatcher.java:75)\n\tat org.neo4j.server.rest.transactional.TransactionalRequestDispatcher.dispatch(TransactionalRequestDispatcher.java:144)\n\tat com.sun.jersey.server.impl.uri.rules.HttpMethodRule.accept(HttpMethodRule.java:302)\n\tat com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)\n\tat com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)\n\tat com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)\n\tat com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1542)\n\tat com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1473)\n\tat com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419)\n\tat com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409)\n\tat com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409)\n\tat com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:558)\n\tat com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:733)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)\n\tat org.neo4j.server.rest.web.CollectUserAgentFilter.doFilter(CollectUserAgentFilter.java:69)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)\n\tat org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)\n\tat org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:95)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:497)\n\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)\n\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)\n\tat org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)\n\tat java.lang.Thread.run(Thread.java:745)\n",
    "message" : "No thread local value in scope for proxy of class com.sun.proxy.$Proxy72"
  } ]
}
```

We have talk about that on slack channel. Maybe is a bug?

Our neo4j config is:

```
dbms.directories.data=/var/lib/neo4j/data
dbms.directories.plugins=/var/lib/neo4j/plugins
dbms.directories.import=import
dbms.security.auth_enabled=false
dbms.connector.bolt.type=BOLT
dbms.connector.bolt.enabled=true
dbms.connector.bolt.tls_level=OPTIONAL
dbms.connector.http.type=HTTP
dbms.connector.http.enabled=true
dbms.connector.http.address=0.0.0.0:7474
dbms.connector.https.type=HTTP
dbms.connector.https.enabled=true
dbms.connector.https.encryption=TLS
dbms.connector.https.address=localhost:7473
dbms.threads.worker_count=5
dbms.logs.http.enabled=true
dbms.logs.query.threshold=0
dbms.logs.query.rotation.keep_number=7
dbms.mode=HA
ha.server_id=2
ha.initial_hosts=10.2.0.140:5001,10.2.0.122:5001,10.2.0.187:5001
ha.host.coordination=10.2.0.140:5001
ha.host.data=10.2.0.140:6001
ha.pull_interval=10
dbms.security.ha_status_auth_enabled=false
dbms.shell.host=0.0.0.0
dbms.unmanaged_extension_classes=*************************
dbms.directories.logs=/var/log/neo4j
dbms.directories.run=/var/run/neo4j
dbms.directories.lib=/usr/share/neo4j/lib
dbms.directories.certificates=/var/lib/neo4j/certificates
```

**Comment by spacecowboy** [comment]
- How big is your cluster?
- Are you writing to the master or a slave? (`127.0.0.1:7474`)

**Comment by mmeloni** [comment]
It's a 3 nodes cluster: 1 Master, 1 slave ed 1 Arbiter.
We have try master and slave, same error
Tnx

**Comment by mmeloni** [comment]
@spacecowboy we would be grateful if you could give us feedback if it's a bug or it is our problem.
T

**Comment by spacecowboy** [comment]
Oh I'm pretty sure it's a bug, but I haven't verified that yet.

**Comment by jimwebber** [comment]
I believe this is an interplay between the batch API and async http logging which me and @nigelsmall are working on. It is not an HA bug (label removed, @spacecowboy)
@mmeloni in the meantime, turn off your HTTP logging and this problem will go away.

**Comment by jimwebber** [comment]
Fixed in de74a0d183aaef16406af7f8f3851ba44d66096f.

**Comment by jimwebber** [comment]
Thank you for taking the time to feedback to us @mmeloni.
